mongo/client_encryption.go

```diff
@@ -72,6 +73,57 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	return ce, nil
 }
 
+// CreateEncryptedCollection creates a new collection with a help of automatic generation of new encryption data keys for null keyIds.
+func (ce *ClientEncryption) CreateEncryptedCollection(ctx context.Context,
+	db *Database, coll string, createOpts *options.CreateCollectionOptions,
+	kmsProvider string, dkOpts *options.DataKeyOptions) (*Collection, error) {
+	if createOpts == nil {
+		return nil, errors.New("nil CreateCollectionOptions")
+	} else if createOpts.EncryptedFields == nil {
```

Comment:
I am not clear on this part. How do we use the `Client.encryptedFieldsMap` in `Database`?

Reply:
Here is one way:

```go
	// Try to get EncryptedFields from options.
	ef := createOpts.EncryptedFields
	if ef == nil {
		// Otherwise, try to get EncryptedFields from EncryptedFieldsMap.
		ef = db.getEncryptedFieldsFromMap(coll)
	}
	if createOpts == nil {
		return nil, errors.New("nil CreateCollectionOptions")
	} else if ef == nil {
		return nil, errors.New("no EncryptedFields defined for the collection")
	}
```